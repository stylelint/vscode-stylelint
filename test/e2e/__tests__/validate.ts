import * as assert from 'node:assert/strict';

import {
	openDocument,
	waitForDiagnostics,
	assertDiagnostics,
	getStylelintDiagnostics,
	executeAutofix,
	closeAllEditors,
} from '../helpers';

describe('"stylelint.validate" setting', () => {
	describe('when set to ["scss"]', () => {
		afterEach(async () => {
			await closeAllEditors();
		});

		it("shouldn't lint or fix css", async () => {
			const { document } = await openDocument('validate/test.css');

			assert.deepEqual(getStylelintDiagnostics(document.uri), []);

			await executeAutofix();

			assert.equal(
				document.getText(),
				`/* prettier-ignore */
a {
  color: #fff;
}
`,
			);
		});

		it('should lint and auto-fix scss', async () => {
			const { document } = await openDocument('validate/test.scss');
			const diagnostics = await waitForDiagnostics(document);

			assertDiagnostics(diagnostics, [
				{
					code: 'color-hex-length',
					codeDescription: 'https://stylelint.io/user-guide/rules/color-hex-length',
					message: 'Expected "#fff" to be "#ffffff" (color-hex-length)',
					range: [2, 9, 2, 13],
					severity: 'error',
				},
			]);

			await executeAutofix();

			assert.equal(
				document.getText(),
				`/* prettier-ignore */
a {
  color: #ffffff;
}
`,
			);
		});

		it("shouldn't lint or fix markdown", async () => {
			const { document } = await openDocument('validate/test.md');

			assert.deepEqual(getStylelintDiagnostics(document.uri), []);

			await executeAutofix();

			assert.equal(
				document.getText(),
				`# title

\`\`\`css
a {
  color: #fff;
}
\`\`\`
`,
			);
		});
	});
});
