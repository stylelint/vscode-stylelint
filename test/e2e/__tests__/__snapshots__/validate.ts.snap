// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"stylelint.validate" setting when set to ["scss"] should lint and auto-fix scss 1`] = `
[
  {
    "code": "color-hex-length",
    "codeDescription": {
      "href": "https://stylelint.io/user-guide/rules/color-hex-length",
    },
    "message": "Expected "#fff" to be "#ffffff" (color-hex-length)",
    "range": {
      "end": {
        "character": 13,
        "line": 2,
      },
      "start": {
        "character": 9,
        "line": 2,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
]
`;

exports[`"stylelint.validate" setting when set to ["scss"] should lint and auto-fix scss 2`] = `
"/* prettier-ignore */
a {
  color: #ffffff;
}
"
`;

exports[`"stylelint.validate" setting when set to ["scss"] shouldn't lint or fix css 1`] = `
"/* prettier-ignore */
a {
  color: #fff;
}
"
`;

exports[`"stylelint.validate" setting when set to ["scss"] shouldn't lint or fix markdown 1`] = `
"# title

\`\`\`css
a {
  color: #fff;
}
\`\`\`
"
`;
